src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -154,6 +154,42 @@ ActorID GcsActorScheduler::CancelOnWorker(const ClientID &node_id,
   return assigned_actor_id;
 }
 
+void GcsActorScheduler::ReleaseUnusedWorkers(
+    const std::unordered_map<ClientID, std::vector<WorkerID>> &node_to_workers) {
+  // The purpose of this function is to release leased workers that may be leaked.
+  // When GCS restarts, it doesn't know which workers it has leased in the previous
+  // lifecycle. In this case, GCS will send a list of worker ids that are still needed.
+  // And Raylet will release other leased workers.
+  // If the node is dead, there is no need to send the request of release unused
+  // workers.
+  const auto &alive_nodes = gcs_node_manager_.GetAllAliveNodes();
+  for (const auto &alive_node : alive_nodes) {
+    const auto &node_id = alive_node.first;
+    nodes_of_releasing_unused_workers_.insert(node_id);
+
+    rpc::Address address;
+    address.set_raylet_id(alive_node.second->node_id());
+    address.set_ip_address(alive_node.second->node_manager_address());
+    address.set_port(alive_node.second->node_manager_port());
+    auto lease_client = GetOrConnectLeaseClient(address);
+    auto release_unused_workers_callback =
+        [this, node_id](const Status &status,
+                        const rpc::ReleaseUnusedWorkersReply &reply) {
+          nodes_of_releasing_unused_workers_.erase(node_id);
+        };
+    auto iter = node_to_workers.find(alive_node.first);
+    if (iter != node_to_workers.end()) {
+      RAY_CHECK_OK(lease_client->ReleaseUnusedWorkers(iter->second,
```

Comment:
Don't use `RAY_CHECK_OK`, because node could fail and this shouldn't fail GCS.

Reply:
Also, better refactor the code structure like this:
```
auto workers_in_use = iter != node_to_workers.end() ? itter->second : {};
lease_client->ReleaseUnusedWorkers();
```

Reply:
fixed